rsqueakvm/model/pointers.py

```diff
@@ -31,94 +29,79 @@ def fillin(self, space, g_self):
         for g_obj in g_self.pointers:
             g_obj.fillin(space)
         pointers = g_self.get_pointers()
-        storage_type = space.strategy_factory.strategy_type_for(pointers, weak=False)  # do not fill in weak lists, yet
+        storage_type = space.strategy_factory.strategy_type_for(self, pointers, weak=False)  # do not fill in weak lists, yet
         space.strategy_factory.set_initial_strategy(self, storage_type,
                                                     g_self.get_class(),
                                                     len(pointers), pointers)
 
     def fillin_weak(self, space, g_self):
         assert g_self.isweak()  # when we get here, this is true
         pointers = self.fetch_all(space)
-        storage_type = space.strategy_factory.strategy_type_for(pointers, weak=True)
+        storage_type = space.strategy_factory.strategy_type_for(self, pointers, weak=True)
         space.strategy_factory.switch_strategy(self, storage_type)
 
     def is_weak(self):
         from rsqueakvm.storage import WeakListStrategy
-        return isinstance(self._get_strategy(), WeakListStrategy)
+        return isinstance(self.strategy, WeakListStrategy)
 
-    def has_class(self):
-        return self.getclass(None) is not None
+    def elidable_strategy(self):
```

Comment:
Since strategies are no longer quasi-immutable, we try to make the reads go away anyway if we have a constant self